build_vm_compilers/lib/src/vm_entrypoint_builder.dart

```diff
@@ -63,16 +64,19 @@ https://github.com/dart-lang/build/blob/master/docs/faq.md#how-can-i-resolve-ski
       }
 
       var transitiveKernelModules = [
-        module.primarySource.changeExtension(vmKernelModuleExtension)
-      ].followedBy(transitiveModules.map(
-          (m) => m.primarySource.changeExtension(vmKernelModuleExtension)));
-      var appContents = <int>[];
-      for (var dependencyId in transitiveKernelModules) {
-        appContents.addAll(await buildStep.readAsBytes(dependencyId));
-      }
-      await buildStep.writeAsBytes(
+        module.primarySource.changeExtension(vmKernelModuleExtension),
+        for (var dep in transitiveModules)
+          dep.primarySource.changeExtension(vmKernelModuleExtension),
+      ];
+      await Future.wait(transitiveKernelModules.map(buildStep.canRead));
+      var appContents = [
+        '#@dill',
+        for (var dependencyId in transitiveKernelModules)
+          p.relative(dependencyId.path, from: buildStep.inputId.path),
```

Comment:
We should use `p.url` here so that the output is hermetic. Hopefully that works in the vm on windows..............

Reply:
We'll need to verify before submitting - I'm not sure if we have a test for that that runs on appveyor...

Reply:
No test running on appveyor 😞